### PR TITLE
Add modal actions to attribute and attribute actions

### LINKED ISSUE #10511: Migrate "Catalog > Attributes & Features > Attributes" listing page
Type: page
Part of Symfony migration project

URL: `/admin-dev/index.php/sell/catalog/attribute-groups`

## Status

Todo:
- [x] Migrate listing https://github.com/PrestaShop/PrestaShop/pull/14040
- [x] Add helper card
- [x] Enable "export" action
- [x] Fix drag-and-drop
- [x] Check that known issues (see below) is fixed
- [x] [Add a modal like on product list (Catalog > Products) to confirm or cancel delete in bulk action](https://github.com/PrestaShop/PrestaShop/issues/14462)
- [x] [Add a modal to confirm or cancel delete action for one row](https://github.com/PrestaShop/PrestaShop/issues/17082)

### Controller / template
- [ ] All controller actions are migrated and protected with `@AdminSecurity` annotations when eligible and `@DemoRestricted` annotations when eligible
- [ ] All links targeting this page are updated: either using _legacy_link or updating the code
- [ ] The new SF routes have `_legacy_link` and `_legacy_parameters` provided and explained
- [ ] Controller catches Domain Exception and translate them nicely
- [ ] Twig templates must contain relevant blocks to allow partial extension by modules
- [ ] JS being used on the page can be used as enabled extensions and use a PageMap for selectors
##### Conventions
- [ ] The new controller actions comply with our naming convention
- [ ] The new SF routes names and paths comply with our naming convention
- [ ] The new Twig template names comply actions with our naming convention
- [ ] Created files (controller, JS assets, CSS assets and twig views) must be in the right folder following the convention
- [ ] CSS being added must follow theme conventions

#### Core
- [ ] Commands and Queries only use native types but internally use VOs
- [ ] All Handlers must have an Interface
##### Conventions
- [ ] The created SF service names comply with our naming convention
- [ ] The created files must follow the folder convention (Domain, Core, Adapter...)

#### Polishing
- [ ] The legacy controller and templates files are gone
- [ ] If PR includes a new concept/mechanism, documentation must be written
- [ ] JS and CSS assets must be compiled and up-to-date
- [ ] License headers must be up-to-date
- [ ] New hooks have been added to the hooks list and the SQL updates
- [ ] Performance must be correct: page loading time, filtering/sorting/searching response time, actions response time

#### Testing
- [ ] Commands and Queries are tested with Behat tests
- [ ] Eligible Core classes are covered with unit tests*
- [ ] Controller is covered with survival test
- [ ] (soon) Controller is covered with E2E tests using puppeeter
- [ ] If we break E2E tests because IDs change, we must update the E2E tests

*Eligible = easy to isolate and unit tests provide values (for example testing getters/setters is useless)

### PR DESCRIPTION
**Pull request by JevgenijVisockij**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add modal actions to attribute and attribute actions
| Type?         | refacto
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #10511.
| How to test?  | Attributes and attribute group deletion (bulk and single) should have modals

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19838)

### CODE REVIEW

File: src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/attribute.yml [hunk 1]
```diff
@@ -52,7 +52,7 @@ admin_attributes_edit:
 
 admin_attributes_delete:
   path: /{attributeId}/delete
-  methods: POST
+  methods: DELETE
```
Comment (PululuK) [inline src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/attribute.yml]: Avoid Break Changes, use ` [POST, DELETE]`

**Review by PululuK — CHANGES_REQUESTED** [review]

File: src/Core/Grid/Definition/Factory/AttributeGridDefinitionFactory.php [hunk 2]
```diff
@@ -24,15 +24,17 @@
  * International Registered Trademark & Property of PrestaShop SA
  */
 
+declare(strict_types=1);
```
Comment (PierreRambaud) [inline src/Core/Grid/Definition/Factory/AttributeGridDefinitionFactory.php]: It's a BC break, you can't add it, this file exists in 1.7.7.x

File: src/Core/Grid/Definition/Factory/AttributeGroupGridDefinitionFactory.php [hunk 3]
```diff
@@ -24,14 +24,14 @@
  * International Registered Trademark & Property of PrestaShop SA
  */
 
+declare(strict_types=1);
```
Comment (PierreRambaud) [inline src/Core/Grid/Definition/Factory/AttributeGroupGridDefinitionFactory.php]: same here

**Review by matks — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @JevgenijVisockij,

The Modal is well displayed.
When I try to delete an item => an exception is displayed
image
https://drive.google.com/file/d/11S8Z4l83Vri-m2pGLB34Q8fDFQw8fGwk/view?usp=sharing

Thanks!

**Comment by JevgenijVisockij** [comment]
> Hi @JevgenijVisockij,
> 
> The Modal is well displayed.
> When I try to delete an item => an exception is displayed
> image
> https://drive.google.com/file/d/11S8Z4l83Vri-m2pGLB34Q8fDFQw8fGwk/view?usp=sharing
> 
> Thanks!

Thanks! Will check that immideatly!

**Comment by JevgenijVisockij** [comment]
Solved the exception issue.

**Review by matks — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @JevgenijVisockij,

It is ok :heavy_check_mark: 
https://drive.google.com/file/d/1WUo4CnkxxRfmjjCh5NVv0Ug2mf1H4kZM/view?usp=sharing

Thanks!

**Comment by Progi1984** [comment]
Thanks @JevgenijVisockij & @khouloudbelguith

**Pull request merged**